setup.py

```diff
@@ -10,7 +10,15 @@
     author='Red Hat',
     version='0.1.0',
     packages=find_packages(include=['receptor_affinity']),
-    install_requires=['click'],
+    install_requires=[
+        'attrs',
+        'click',
+        'pyparsing',
+        'pyyaml',
+        'receptor',
```

Comment:
Doesn't this create a circular dependency?

Reply:
Hmm.

```console
$ git grep receptor
.github/workflows/package.yml:        python -c 'import receptor_affinity; print(receptor_affinity.__version__)'
README.md:Set of tools to check receptor's binding affinity.
receptor_affinity/__init__.py:    __version__ = get_distribution('receptor-affinity').version
receptor_affinity/debugger.py:from receptor import Controller
receptor_affinity/debugger.py:from receptor import ReceptorConfig
receptor_affinity/debugger.py:from receptor.exceptions import UnrouteableError
receptor_affinity/debugger.py:        self.data_path = data_path or f"/tmp/receptor/{self.uuid}"
receptor_affinity/debugger.py:        self.listen = listen or f"receptor://127.0.0.1:{random_port()}"
receptor_affinity/debugger.py:        return web.Response(text=json.dumps(self.controller.receptor.router.get_edges()))
receptor_affinity/debugger.py:        peer = request.query.get("peer", "receptor://127.0.0.1:8889")
receptor_affinity/debugger.py:            while not self.controller.receptor.router.node_is_known(recipient) and (
receptor_affinity/mesh.py:            self.data_path = f"/tmp/receptor/{str(self.uuid)}"
receptor_affinity/mesh.py:            self.listen = f"receptor://0.0.0.0:{random_port()}"
receptor_affinity/mesh.py:            listen=config.get("listen", f"receptor://0.0.0.0:{random_port()}"),
receptor_affinity/mesh.py:            st = ["python", "-m", "cProfile", "-o", f"{self.name}.prof", "-m", "receptor.__main__"]
receptor_affinity/mesh.py:            st = ["receptor"]
receptor_affinity/mesh.py:                name="diag_node", listen=f"receptor://127.0.0.1:{random_port()}"
receptor_affinity/mesh.py:                listen=f"receptor://127.0.0.1:{controller_port}",
receptor_affinity/mesh.py:                    listen=f"receptor://127.0.0.1:{random_port()}",
receptor_affinity/ping.py:from receptor import Controller
receptor_affinity/ping.py:from receptor import ReceptorConfig
receptor_affinity/ping.py:        while not controller.receptor.router.node_is_known(config.ping_recipient) and (
receptor_affinity/ping.py:@click.option("--data-path", default="/tmp/receptor")
receptor_affinity/ping.py:@click.option("--peer", default="receptor://127.0.0.1:8889")
setup.py:    name='receptor-affinity',
setup.py:    packages=find_packages(include=['receptor_affinity']),
setup.py:        'receptor',
setup.py:        "Set of tools to check receptor's binding affinity."
setup.py:            'affinity = receptor_affinity.cli:cli'
setup.py:    url='https://github.com/project-receptor/affinity',
tests/test_affinity.py:import receptor_affinity
tests/test_affinity.py:    assert hasattr(receptor_affinity, '__version__')
```

The fact that we're importing receptor code implies that there really is a circular dependency. That's a problem, no?

Reply:
I don't think this is a circular dependency since receptor itself does not depend on affinity. Only receptor's tests.

Affinity has the ability of spawning a node on the mesh so it can do some actions. To do that we have two options here not list receptor and assume it will be installed (like the receptor under test) or enforce receptor is available by listing it as a requirement. I went for the later option to ensure receptor will be pulled when running affinity standalone.

Reply:
The dependency tree looks like this, then?

![out](https://user-images.githubusercontent.com/1012989/73190090-7016e280-40f3-11ea-8c82-66305a8f8e2f.png)

Reply:
It would be more like so: `receptor tests` -> `affinity` -> `receptor`. We are not import anything from receptor on the tests (see below). We are actually running receptor as a cmd line tool and making assertions based on its output and the debug node that affinity introduces to the network.

```console
$ cd receptor/tests/perf

$ ag "import receptor"

$ ag "from receptor"   
test_route.py
1:from receptor_affinity.mesh import Node
2:from receptor_affinity.mesh import Mesh
3:from receptor_affinity.utils import random_port

test_websockets.py
1:from receptor_affinity.mesh import Node
2:from receptor_affinity.mesh import Mesh
3:from receptor_affinity.utils import random_port

test_ping.py
1:from receptor_affinity.mesh import Mesh
```

Reply:
Yes.